KOGITO-9460: Create a new guide describing the workflow CLI (#432)
kiegroup/kogito-docs: serverlessworkflow/modules/ROOT/pages/testing-and-troubleshooting/kn-plugin-workflow-overview.adoc

.Example output
[source,text]
----

> Manage SonataFlow projects

Author: this is an output of the CLI, need to check if productized has any changes

----
--

[[proc-build-sw-project-kn-cli]]
== Running a workflow project using Knative CLI

After creating your workflow project, you can use the `run` command with `kn workflow` to build & run your workflow project in your current directory. 

This will start a {product_name} docker image and map your local folder to this image.

.Prerequisites
* {product_name} plug-in for Knative CLI is installed.
+
For more information about installing the plug-in, see <<proc-install-sw-plugin-kn-cli, Installing the {product_name} plug-in for Knative CLI>>.

* A workflow project is created.
+
For more information about creating a workflow project, see <<proc-create-sw-project-kn-cli, Creating workflow project using Knative CLI>>.
* Minikube cluster is running locally.

.Procedure
. In Knative CLI, enter the following command to build and run your workflow project:
+
--

> .Run the project and start a local development image`

Reviewer: ```suggestion
.Run the project and start a local development image.
```

[source,shell]
----
kn workflow run
----
--

> . Navigate to `localhost:8080/q/dev` to access the Development UI locally.

Reviewer 2: By default, the `localhost:8080/q/dev` should pop up automatically in the default web browser once it is ready (unless `--open-dev-ui=false` is specified).

So maybe something like this would be more appropriate:
```suggestion
. Once the project is ready, the Development UI will be opened up in a browser automatically (on `localhost:8080/q/dev`).
```

[[proc-deploy-sw-project-kn-cli]]
== Deploying a workflow project using Knative CLI

You can use the `deploy` command combined with `kn workflow` to deploy your workflow project in your current directory. 

.Prerequisites
* {product_name} plug-in for Knative CLI is installed.
+
For more information about installing the plug-in, see <<proc-install-sw-plugin-kn-cli, Installing the {product_name} plug-in for Knative CLI>>.

* A workflow project is created.
+
For more information about created a workflow project, see <<proc-create-sw-project-kn-cli, Creating workflow project using Knative CLI>>.

* A minikube cluster is running locally.

.Procedure
. In Knative CLI, enter the following command to deploy your workflow project:
+
--
.Deploy a workflow project
[source,shell]
----
kn workflow deploy
----

.Deploy a workflow project using `--path`
[source,shell]
----

> kn workflow deploy --path other-path

Reviewer 2: The `--path` flag is only within `quarkus deploy`. The `deploy` takes another flags:
```
Flags:
  -h, --help                        help for deploy
  -c, --manifestPath string         Target directory of your generated Kubernetes manifests.
  -n, --namespace string            Target namespace of your deployment.
  -s, --supportFilesFolder string   Specify a custom support files folder
```

Reviewer 2: And maybe keep the `[-n|--name]` system of mentioning flags that is used later bellow 😉

Author: Thanks fixing

Author: Ok I removed this, and added mention of help. We can improve with examples for each command in follow-up PR

----

Also, ensure that you have access to your cluster and your cluster can access the generated container image.

[NOTE]
====
You can use the `kubectl` command line if you want to use a complex deployment setup for your workflow project.
====
--

[[proc-create-quarkus-sw-project-kn-cli]]
== Creating a Quarkus Workflow project using Knative CLI

After installing the {product_name} plug-in, you can use the `quarkus create` command with `kn workflow` to scaffold a new Quarkus Workflow project in your current directory.

> The `quarkus create` command sets up {product_name} project containing minimal extensions to build a workflow project. The `create` command sets up Quarkus project containing minimal extensions to build a workflow project. Also, the generated workflow project contains a "hello world" `workflow.sw.json` file in your `./<project-name>/src/main/resources` directory.

Reviewer: ```suggestion
The `quarkus create` command sets up {product_name} project containing minimal extensions to build a workflow project. The `create` command sets up a Quarkus project containing minimal extensions to build a workflow project. Also, the generated workflow project contains a "hello world" `workflow.sw.json` file in your `./<project-name>/src/main/resources` directory.
```

endif::[]

.Procedure
. In Knative CLI, enter the following command to create a new project:
+
--
.Creates a project named `new-project`
[source,shell]
----
kn workflow quarkus create
----

By default, the generated project is named as `new-project`. You can overwrite the project name by using the `[-n|--name]` flag as follows:

.Create a project named `my-project`
[source,shell]
----
kn workflow quarkus create --name my-project
----
--

. Add more extensions to the Quarkus project during its creation by using the `[-e|--extension]` flag as follows:
+
--
.Create a project with `quarkus-jsonp and quarkus-smallrye-openai` extensions
[source,shell]
----
kn workflow quarkus create --extension quarkus-jsonp,quarkus-smallrye-openapi
----
You can add multiple extensions using the comma-separated names of the extensions in the previous command.
[NOTE]
====
When you run the `create` command for the first time, it might take a while due to the necessity of downloading the required dependencies for the Quarkus project.
====
--

[[proc-build-quarkus-sw-project-kn-cli]]
== Building a Quarkus workflow project using Knative CLI

> After creating your workflow project, you can use the `quarkus build` command with `kn workflow` to build your workflow project in your current directory and to generate a container image. 

Reviewer: ```suggestion
After creating your workflow project, you can use the `quarkus build` command with `kn workflow` to build your workflow project in your current directory and to generate a container image. 
```
